provider/pkg/clients/clients.go

```diff
@@ -164,23 +166,25 @@ func (dcs *DynamicClientSet) searchKindInGVResources(gvResources *v1.APIResource
 // IsNamespacedKind checks if a given GVK is namespace-scoped. Known GVKs are compared against a static lookup table.
 // For GVKs not in the table, attempt to look up the GVK from the API server. If the GVK cannot be found, a
 // NoNamespaceInfoErr is returned.
-func IsNamespacedKind(gvk schema.GroupVersionKind, clientSet *DynamicClientSet) (bool, error) {
+func IsNamespacedKind(gvk schema.GroupVersionKind, disco discovery.DiscoveryInterface) (bool, error) {
```

Comment:
Rationale: changing the function signature to make it more easily testable (using the fake disco client).